### PR TITLE
Fix migrations failing when no statuses exist and other various migrations fixes

### LINKED ISSUE #272: Migration Can Fail if Status Doesn't Exist
https://github.com/nautobot/nautobot-app-firewall-models/blob/5b71b14632726da759251069945d0287018e6f49/nautobot_firewall_models/utils/__init__.py#L14-L15

If the configured default status or the defaulted value does not exist, the migration that sources this function can fail.

### PR DESCRIPTION
**Pull request by gsnider2195**
Closes #272 

- In migrations only create statuses that are configured in the plugin settings `default_status` and `allowed_status` if they don't already exist. Associate all firewall models that have a status field to these statuses through the `content_types` field.
- In database_ready signal, create statuses that are configured in the plugin settings `default_status` and `allowed_status`. Associate all firewall models that have a status field to these statuses through the `content_types` field.
- In database_ready signal, associate all firewall models that have a status field to the `Active`, `Staged` and `Decommissioned` statuses. Skips any of these statuses if they don't already exist. This logic was removed from multiple migrations

### CODE REVIEW

File: nautobot_firewall_models/utils/__init__.py [hunk 1]
```diff
@@ -2,17 +2,84 @@
...
+from nautobot_firewall_models.constants import PLUGIN_CFG
+
+
+def _create_status(status_name, apps=django_apps):
+    """Create a status with the given name, using nautobot default description and color if applicable."""
+    Status = apps.get_model("extras.Status")  # pylint: disable=invalid-name
+    defaults = {"description": STATUS_DESCRIPTION_MAP.get(status_name, "")}
+    if status_name in STATUS_COLOR_MAP:
+        defaults["color"] = STATUS_COLOR_MAP[status_name]
+    status, _ = Status.objects.get_or_create(name=status_name, defaults=defaults)
```
Comment (itdependsnetworks) [inline nautobot_firewall_models/utils/__init__.py:25]: What happens if the status "active" exists and the default "Active" is there?

File: nautobot_firewall_models/migrations/0002_custom_status.py [hunk 2]
```diff
@@ -2,69 +2,76 @@
...
     dependencies = [
+        ("contenttypes", "0002_remove_content_type_name"),
         ("extras", "0033_add__optimized_indexing"),
         ("nautobot_firewall_models", "0001_initial"),
     ]
 
     operations = [
         migrations.RunPython(code=create_status, reverse_code=reverse_create_status),
-        migrations.RunPython(code=create_default_objects, reverse_code=reverse_create_default_objects),
+        migrations.RunPython(code=create_default_objects, reverse_code=migrations.RunPython.noop),
```
Comment (smk4664) [inline nautobot_firewall_models/migrations/0002_custom_status.py:76]: These objects would get deleted when reversing to `Zero` anyways, I believe this is acceptable and the `reverse_create_default_objects` function can be removed.

File: nautobot_firewall_models/utils/__init__.py (hunk 1 above)
Comment (gsnider2195) [reply nautobot_firewall_models/utils/__init__.py:25]: It's case sensitive so it will choose whichever status matches.

**Review by smk4664 — APPROVED** [review]
This looks great and resolves the issue with migrations.

**Pull request merged**